service/TransactionsServiceImpl.java

```diff
@@ -8,26 +8,50 @@
 import com.app.utils.Constants;
 import com.app.utils.ValidationUtils;
 import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
 import java.math.BigDecimal;
 import java.math.RoundingMode;
 import java.time.LocalDateTime;
+import java.time.temporal.TemporalUnit;
 import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 
 @Service
 public class TransactionsServiceImpl implements TransactionsService {
 
-    private static final org.slf4j.Logger LOGGER= LoggerFactory.getLogger(TransactionsServiceImpl.class);
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(TransactionsServiceImpl.class);
+    private static final Collection<Transaction> transactions = Collections.synchronizedCollection(new ArrayList<Transaction>());
+    private static final StatisticsDTO statisticsDTO = new StatisticsDTO();
 
+    @Override
+    public synchronized StatisticsDTO getStatistics() {
 
-    private List<Transaction> transactions = new ArrayList<Transaction>();
-
+        return statisticsDTO;
+    }
 
     @Override
-    public StatisticsDTO getStatistics() {
+    public void saveTransaction(TransactionDTO transactionDTO) throws NotFoundException, UnprocessableException {
```

Comment:
I think NotFoundException and UnprocessableExcetiopns are unchecked, why do you need to add them here?